src/compile/data/filterinvalid.ts

```diff
@@ -0,0 +1,68 @@
+import {isScaleChannel} from '../../channel';
+import {field as fieldRef, FieldDef} from '../../fielddef';
+import {hasContinuousDomain, ScaleType} from '../../scale';
+import {Dict, extend, keys} from '../../util';
+import {VgTransform} from '../../vega.schema';
+import {ModelWithField} from '../model';
+import {DataFlowNode} from './dataflow';
+
+export class FilterInvalidNode extends DataFlowNode {
+  private fieldDefs: Dict<FieldDef<string>>;
+
+  public clone() {
+    return new FilterInvalidNode(extend({}, this.fieldDefs));
+  }
+
+  constructor(fieldDefs: Dict<FieldDef<string>>) {
+   super();
+
+   this.fieldDefs = fieldDefs;
+  }
+
+  public static make(model: ModelWithField): FilterInvalidNode {
+
+    const filter = model.reduceFieldDef((aggregator: Dict<FieldDef<string>>, fieldDef, channel) => {
```

Comment:
Comparing to the old `nullfilter.ts`, somehow we no longer check `model.config.invalidValues === 'filter'` here.  
Let's add it back; otherwise we can't disable invalid filter.